File: squares-of-a-sorted-array/squares-of-a-sorted-array.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        int left = 0;
        int right = nums.Length-1;
        var res = new int[nums.Length];
        
        for(int i = nums.Length-1; i >=0; i--)
        {
            if(Math.Abs(nums[left]) >= Math.Abs(nums[right]))
            {
                res[i] = nums[left] * nums[left];
                left++;
            } else {
                res[i] = nums[right] * nums[right];
                right--;
            }
            
        }
        
        return res;
        
    }
}

File: squares-of-a-sorted-array/squares-of-a-sorted-array.ts
function sortedSquares(nums: number[]): number[] {
    const ret: number[] = [];
    let left: number = 0;
    let right: number = nums.length-1;
    
    for(let i = nums.length-1; i >= 0; i--){
        if(Math.abs(nums[right]) > Math.abs(nums[left])){
            ret[i] = nums[right] * nums[right];
            right--;
        } else {
            ret[i] = nums[left] * nums[left];
            left++;
        }
    }
    return ret;
};